packages/react-components/tsconfig.lib.json

```diff
@@ -0,0 +1,25 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "noEmit": false,
+    "lib": ["ES2019", "dom"],
+    "outDir": "dist",
+    "declaration": true,
+    "types": ["static-assets", "environment", "inline-style-expand-shorthand"]
+  },
+  "exclude": [
+    "**/*.spec.ts",
+    "**/*.spec.tsx",
+    "**/*.test.ts",
+    "**/*.test.tsx",
+    "**/*.spec.js",
+    "**/*.spec.jsx",
+    "**/*.test.js",
+    "**/*.test.jsx",
+    "**/*.stories.ts",
+    "**/*.stories.tsx",
+    "**/*.stories.js",
+    "**/*.stories.jsx"
+  ],
+  "include": ["./src/**/*.ts", "./src/**/*.tsx", "./src/**/*.js", "./src/**/*.jsx"]
```

Comment:
@Hotell this package is a third option, it's a mix of JS & TS 🥲

Reply:
![image](https://user-images.githubusercontent.com/14183168/143591457-f59c59dd-f18b-49e7-b606-5504bb9494f1.png)

Tested on latest changes from `master`. Manual changes to config are still required.

Reply:
it is not, please re-run the generator or remove manually all `js` globs

Reply:
I applied your suggestions, from latest build:

![image](https://user-images.githubusercontent.com/14183168/143596482-ed1cd1bc-f710-423c-b10c-25796fff2ffe.png)

Reply:
This happens because `packages/react-components/.storybook/tsconfig.json` includes all JS files:

![image](https://user-images.githubusercontent.com/14183168/143596715-7fe618fb-6389-4d36-9f87-6b2e0e8c1287.png)

Reply:
Simple solution is to make `main.utils.test.js` to `main.utils.test.ts` 😅

Reply:
looks like we write too often at the same time :D 

https://github.com/microsoft/fluentui/pull/20749#discussion_r757595730

Reply:
> This happens because packages/react-components/.storybook/tsconfig.json includes all JS files

that's on purpose, to get proper type checking / DX